pkg/providers/aws/provider_postgres.go

```diff
@@ -129,6 +132,32 @@ func (p *AWSPostgresProvider) CreatePostgres(ctx context.Context, pg *v1alpha1.P
 	// setup aws RDS instance sdk session
 	rdsSession := createRDSSession(stratCfg, providerCreds)
 
+	// Retrieve service maintenance updates, create and export Prometheus metrics
```

Comment:
What do you think about we abstract this logic to separate functions. These create functions are becoming more and more bloated.